Add support for intel-quark ZE250

telphan opened the pull request:
@agherzan @floion

Commit:
.gitignore: Add file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
README.md: Add file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
LICENSE: Add file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
VERSION: Release board on 2.0.0-beta.4

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
intel-quark.cofee: Add file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
.gitmodules: Add file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
samples: Add sample configuration files

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
README.md: Add file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
COPYING.Apache-2.0: Add file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
linux-yocto: Configure kernel for resin use

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
resin-init-flasher: Configure the flasher type

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
resin-image: Configure images to our board needs

There is a need for an initramfs in our flasher type to allow
triggering and settling of udev, to discover the necessary information
about the connected usb drive.

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
initrdscript: Provide initramfs script to be able to detect partitions inside the USB stick

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
docker-resin-supervisor-disk: Provide supervisor information

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
grub-efi: Provide grub configuration files for internal and external use

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
layer.conf: Add file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
go-cross: Compile go for intel-quark's architecture

Intel quark processors lack support for SIMD instruction set (e.g. MMX
and SSE). As go requires these instruction for atomic operations we will
replace them with equivalent instructions.

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
docker: Compile docker for this architecture

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
Add required submodules for the project

Signed-off-by: Theodor Gherzan <theodor@resin.io>

agherzan commented on .gitignore:
```diff
@@ -0,0 +1,15 @@
+*.json
+
+# dynamically created by the build script
+conf-notes.txt
```
Please add a new line

telphan replied:
Ok.

agherzan commented:
Typo: "resin-init-flasher: Configure the flsaher type"

agherzan commented on layers/meta-resin-intel-quark/recipes-core/images/resin-image.bbappend:
```diff
@@ -0,0 +1,6 @@
+include resin-image.inc
+
+RESIN_BOOT_PARTITION_FILES += " \
+    core-image-minimal-initramfs-intel-quark.cpio.gz:/initramfs \
```
Could you please add some explanation why we need an initramfs?

telphan replied:
I will add an explanation.

agherzan commented on layers/meta-resin-intel-quark/recipes-core/initrdscripts/files/init-external-media.sh:
```diff
@@ -0,0 +1,125 @@
+#!/bin/sh
+
+PATH=/sbin:/bin:/usr/sbin:/usr/bin
```
Isn't this the same script used for nuc? Can we abstract this into a meta-resin feature? As well, why do we need the initramfs in normal images? Does it boot from usb or from the internal device?

telphan replied:
This is not the same script as the nuc, chances are this might work for nuc too, but not the other way around. All testing was done on usb, but the device should boot from internal medium. I will try to see if I can get rid of the initramfs from emmcs.

agherzan commented on layers/meta-resin-intel-quark/recipes-containers/docker-disk/docker-resin-supervisor-disk.bbappend:
```diff
@@ -0,0 +1,2 @@
+SUPERVISOR_REPOSITORY = "resin/i386-supervisor"
```
CC https://github.com/resin-os/resinos/issues/102

agherzan commented:
Typo: "grub-efi: Provide frub configuration files for internal and external use"

agherzan commented on layers/meta-resin-intel-quark/recipes-devtools/go/files/0002-implement-atomic-quadword-ops-with-FILD-FISTP.patch:
```diff
@@ -0,0 +1,91 @@
+From: Petros Angelatos <petrosagg@gmail.com>
+Date: Tue, 18 Oct 2016 08:13:44 -0700
+Subject: [PATCH 2/2] implement atomic quadword ops with FILD/FISTP
+
```
Lacking upstream status tag.

telphan replied:
Fixing.

agherzan commented on layers/meta-resin-intel-quark/recipes-devtools/go/files/0001-Revert-runtime-check-and-fail-early-with-a-message-i.patch:
```diff
@@ -0,0 +1,81 @@
+Date: Thu, 20 Oct 2016 20:20:56 -0700
+Subject: [PATCH 1/2] Revert "runtime: check and fail early with a message if
+ MMX is not available on 386"
+
```
Lacking upstream status tag.

telphan replied:
Fixing.

agherzan commented on layers/meta-resin-intel-quark/recipes-devtools/go/go-cross_%.bbappend:
```diff
@@ -0,0 +1,10 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+
+SRC_URI += " \
+    file://0001-Revert-runtime-check-and-fail-early-with-a-message-i.patch \
```
I'm wondering if these patches should not go in meta-resin and then only activate them if mmx is not available on the board. In this way multiple boards in the future will be able to benefit from it. @floion , what do you think?

agherzan commented:
"layers: Add supporting layers (i.e. submodules)" should be rephrased as it doesn't only add layers (see resin-yocto-scripts).

Commit:
meta-resin: Update layer to v2.0-beta.4 so we can include the new supervisor

Some fixes are needed from our common layer. e.g. Expanding the data
partition is broken on previous version.

Signed-off-by: Theodor Gherzan <theodor@resin.io>

Commit:
intel-quark.svg: Add icon file

Signed-off-by: Theodor Gherzan <theodor@resin.io>

agherzan commented:
This needs to be blocked until supervisor is fixed for x87.

michal-mazurek commented on layers/meta-resin-intel-quark/README.md:
```diff
@@ -0,0 +1,4 @@
+# Resin.io layer for the intel-quark board
+
+## Description
+This repository enables building resin.io for intel-quark
```
"resin.io" should be upper case

telphan replied:
I have only seen it written in lower case. Marked as NIT
